### ISSUE TITLE
What's Directv Go detect? 😥

### ISSUE THREAD

**Issue by Triviox** [adrianmmiller/Hide-My-Applist-Guide#5]
Hello! I have a Redmi 9t with magisk 26.1, magiskhide 1.10.3, shamiko 0.7.1, Universal Safetynet fix 2.4.0 Mod displax, lsposed zydisk 1.8.6, HMA 3.1.1.
Pass Safetynet, the applist detector and 2 of 3 in integrity checker,,, but Directv Go still detect non-stock-system. I'm in stock rom with stock recovery.
Dime idea where is the problem?
Thanks! 💪

![Screenshot_2023-05-08-13-14-36-246_com tsng hidemyapplist](https://user-images.githubusercontent.com/19431071/236877147-4318701e-80b1-4b8f-b684-598039c78fc6.jpg)
![Screenshot_2023-05-08-13-14-24-066_com tsng hidemyapplist](https://user-images.githubusercontent.com/19431071/236877168-440d7880-6cfa-4169-95e0-44bae276ba4c.jpg)
![Screenshot_2023-05-08-13-13-27-407_gr nikolasspyr integritycheck](https://user-images.githubusercontent.com/19431071/236877170-1217bd35-ad1a-498a-8985-d303ee1a2d2f.jpg)
![Screenshot_2023-05-08-13-13-16-251_rikka safetynetchecker](https://user-images.githubusercontent.com/19431071/236877173-f7e21d34-cb29-41ab-aa8f-4f4bddbdaf51.jpg)
![Screenshot_2023-05-08-13-13-08-944_icu nullptr applistdetector](https://user-images.githubusercontent.com/19431071/236877178-4ecd0bbc-d94b-4097-b5f7-a6ed764ccab6.jpg)
![Screenshot_2023-05-08-13-13-00-231_com directv dtvlatam](https://user-images.githubusercontent.com/19431071/236877180-6bf992a5-6f66-400b-882e-e85dbfeb1e43.jpg)

**Comment by adrianmmiller** [comment]
Hi,

Sadly not every every app will magically work with HMA if its somehow still detecting root (or a change to the Operating System) via other base methods.

Im currently testing KernelSU, NOT magisk (for a change), and KernelSU has root in the kernel, and therefore does expose root with quite the same method of ease as magisk does.

When i tested the app (i had to download via apkmirror as it was nto available via Play in Australia) it first complained about me having USB Debugging on (este dispotivo esta con el modo depuracion USB activado), once that was disabled, i got the same error you did. Applying the root blakclist to the app, rebooting and clearing the apps data before testing again made no change.

So, sorry but i cant help here as it appears to be detecting root via some method not generally blocked by basic root hiding.

Ordinarily i would be using Magisk and Shamiko, so all i can suggest is try the following, which i would be trying if using Magisk and not testing KernelSU this week:

Basic check:

* Do you use a custom recovery like TWRP? If Yes, rename or delete the /sdcard/TWRP folder (backing up and saved backups if they exists first!.....some apps use this folder to detect root, though its unliekly as im on a Pixel and theres no chance of custom recovery, this is a basic checklist)

Deny List:

Check you have the app in Magisks inbuilt Deny List (HMA is an additojnal tool, you still need to Deny ( "hide") apps via Deny List in Magisk App

* In Magisk App -> Settings -> Deny List, add Direct TV go to the Deny List
* Reboot, clear the data for Direct TV Go and test it

If that fails:

* Download Shamiko from here: https://github.com/LSPosed/LSPosed.github.io/releases
* Flash Shamiko module via Magisk App 
* Still in Magisk App -> Settings -> Disable "Enforce Deny List" (this step is needed when running Shamiko)
* Reboot, clear the data for Direct TV Go and test it

You may still fail even with all these, if thats the case, there may be upcoming Magisk changes which may provide better hiding methods. We just try the best we can with this stuff....
 
Note: i noticed in your 1st screenshot, the root blacklist you have Applits Detector selectd (and its also selected in the apps the blacklist is applied to. You dont need it selected in the root blacklist :)

**Comment by Triviox** [comment]
Hello Adrian! Thanks for your clear and detailed answer!
About your questions:
- Im not using TWRP rigth now, i made a clear install with "flash all" to detele all "trash" about olders twrp, magisk, xposed, etc.
- I didnt post it, but Directv Go is in Magisk Deny List.
- I have Shamiko module installed, without "enforced deny list".
- I clear data in Directv Go app, but still detect i have a modified system

Them, will wait for Magisk 27.x, i read a lot about good news about the next udpdate.
Thanks again for your time, patience and clear "how to" and answer.
Regard! 

PS: Oks, will remove App list derector fom HMA hehe.

**Comment by adrianmmiller** [comment]
Cheers for clarifying you had already done some of my suggestions

If you monitor the Magisk threads, and especially posts by pndwal (also a contributor to the revamped HMA Guide, launching soon), you will be keeping up to date, as he posts the most about whats happening in magisk land. pndwal is already testing the early changes coming for the next version of magisk, and posts of all changes and releases about magisk
